### PR TITLE
CNV-44913: missing data source selection

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

Datasource modal was missing so the disk with DataSource source was displaying the OtherDiskModal that does not contain info about source. This also let the user change datasource as in previous versions

## 🎥 Demo

**Before**
Screenshot 2024-09-06 at 17 16 39

**After**

Screenshot 2024-09-06 at 17 13 03

### CODE REVIEW

File: src/utils/hooks/useNamespacesResources.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react';
+
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type useNamespacedResourcesType = <T extends K8sResourceCommon>(
+  namespace: string,
+  groupVersionKind: { group: string; kind: string; version: string },
+) => [T[], boolean, any];
+
+const useNamespacedResources: useNamespacedResourcesType = <T extends K8sResourceCommon>(
```
Comment (pcbailey) [inline src/utils/hooks/useNamespacesResources.ts:10]: The file name and component name aren't consistent. I assume you meant for the file name to `useNamespacedResources`, too?

File: src/utils/hooks/useNamespacesResources.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        namespaced: true,
+      }
+    : null;
+
+  const [unsortedResources, loaded, error] = useK8sWatchResource<T[]>(watchResources);
+
+  const resources = useMemo(
+    () =>
+      (unsortedResources || [])?.sort((a, b) =>
+        a?.metadata?.name?.localeCompare(b?.metadata?.name),
```
Comment (pcbailey) [inline src/utils/hooks/useNamespacesResources.ts:28]: I suggest using the `getName` function here.

File: src/utils/components/DiskModal/components/utils/selectors.ts [hunk 3]
```diff
@@ -24,3 +24,9 @@ export const getErrorSnapshotNamespace = (errors: FieldErrorsImpl<V1DiskFormStat
 
 export const getDataVolumeTemplateSize = (diskState: V1DiskFormState): string =>
   diskState.dataVolumeTemplate?.spec?.storage?.resources?.requests?.storage;
+
+export const getErrorDataSourceName = (errors: FieldErrorsImpl<V1DiskFormState>): FieldError =>
```
Comment (pcbailey) [inline src/utils/components/DiskModal/components/utils/selectors.ts:28]: Consider `getDataSourceNameError` and `getDataSourceNamespaceError` instead.

File: src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceDataSourceSelect/DiskSourceDataSourceSelectNamespace.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    <Controller
+      render={({ field: { onChange, value } }) => (
+        <FormGroup
+          fieldId={diskSourceDataSourceNamespaceFieldID}
+          id={diskSourceDataSourceNamespaceFieldID}
+          isRequired
+          label={t('DataSource project')}
+        >
+          <InlineFilterSelect
+            options={projectNames?.map((name) => ({
```
Comment (metalice) [inline src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceDataSourceSelect/DiskSourceDataSourceSelectNamespace.tsx:43]: i would put it in a useMemo , as other things from from hook can cause it to rerun

File: src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceDataSourceSelect/DiskSourceDataSourceSelectName.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    <Controller
+      render={({ field: { onChange, value } }) => (
+        <FormGroup
+          fieldId={diskSourceDataSourceNameFieldID}
+          id={diskSourceDataSourceNameFieldID}
+          isRequired
+          label={t('DataSource name')}
+        >
+          <InlineFilterSelect
+            options={dataSourceNames?.map((name) => ({
```
Comment (metalice) [inline src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceDataSourceSelect/DiskSourceDataSourceSelectName.tsx:54]: cna be in useMemo to reduce rendering

**Pull request closed without merging**